lib/index.js

```diff
@@ -170,6 +170,119 @@ class VideoAsset extends BaseAsset {
 
 exports.VideoAsset = VideoAsset;
 
+class BlogArticle extends BaseAsset {
+    constructor() {
+        super();
+        this._object_type = "ArticleObject";
+        this._content_type = "text/html";
+    }
+
+    set_author(value) { this._author = value; }
+    set_date_published(value) { this._date_published = value; }
+    set_main_image(value) { this._main_image = value; }
+    set_main_image_caption(value) { this._main_image_caption = value; }
+    set_body(value) { this._body = value; }
+    set_read_more_text(value) { this._read_more_text = value; }
+    set_tags (value) { this._tags = value; }
+    set_as_static_page() { this._ekn_tags.push(EKN_STATIC_TAG); }
+
+    to_metadata() {
+        const metadata = super.to_metadata();
+        metadata['document'] = this._document.html();
+        return metadata;
+    }
+
+    _process() {
+        this._render();
+        return Promise.resolve().then(() => {
+            const $elem = this._document;
+
+            $elem('img, video, audio').each(function() {
+                const $asset = cheerio(this);
+
+                const asset_id = $asset.attr('data-libingester-asset-id');
+                if (asset_id) {
+                    $asset.attr('data-soma-job-id', asset_id);
+                    $asset.attr('data-libingester-asset-id', null);
+                }
+
+                if (!$asset.attr('data-soma-job-id')) {
+                    throw new ValidationError("Document has media without associated asset. Please ingest it or drop it.");
+                }
+            });
+        });
+    }
+
+    _render() {
+        let main_image;
+        if (this._main_image)
+            main_image = `<img data-libingester-asset-id="${this._main_image.asset_id}">`;
+
+        let main_image_caption;
+        if (this._main_image_caption)
+            main_image_caption = `<div class="main-image-caption">${this._main_image_caption}</div>`;
+
+        let template =`
+<header>
+    <div class="categories" id="categories-links"></div>
+    <div class="extra-header-right">
+        <span class="authors">${this._author}</span>
+        &mdash;
+        <span class="date-published">${this._date_published}</span>
+    </div>
+</header>
+<article>
+    <div class="article-title">
+        <h1>${this._title}</h1>
+    <div>
+    <div class="main-image">
```

Comment:
I had it set up so that there is no surrounding element if the main image was missing.